mmdet3d/datasets/scannet_dataset.py

```diff
@@ -126,3 +130,362 @@ def show(self, results, out_dir, show=True):
+            # are the new label ids.
+            # used for changing pixel labels in load_annotations.
+            self.label_map = {
+                cls_id: self.ignore_index
+                for cls_id in self.ALL_CLASS_IDS
+            }
+            self.label_map.update(
+                {cls_id: i
+                 for i, cls_id in enumerate(self.VALID_CLASS_IDS)})
+            self.label2cat = {
+                i: cat_name
+                for i, cat_name in enumerate(class_names)
+            }
+
+        palette = [
+            self.PALETTE[self.CLASSES.index(cls_name)]
+            for cls_name in class_names
+        ]
+
+        return class_names, palette
+
+    def _get_room_idxs_and_label_weight(self, room_idxs, label_weight):
+        """Compute room_idxs for data sampling and label weight for loss \
+        calculation.
+
+        We sample more times for rooms with more points. Label_weight is
+        inversely proportional to number of class points.
+        """
+        if self.test_mode:
+            return np.array(range(len(self.data_infos))), None
+
+        if room_idxs is not None:  # precomputed
+            if isinstance(room_idxs, str):
+                room_idxs = np.load(room_idxs).astype(np.int32)
+            if isinstance(label_weight, str):
+                label_weight = np.load(label_weight).astype(np.float32)
+            return room_idxs, label_weight
+
+        num_classes = len(self.CLASSES)
+        num_point_all = []
+        label_weight = np.zeros((num_classes + 1, ))  # ignore_index
+        for data_info in self.data_infos:
+            label = self._convert_to_train_label(
+                osp.join(self.data_root, data_info['pts_semantic_mask_path']))
+            num_point_all.append(label.shape[0])
+            class_count, _ = np.histogram(label, range(num_classes + 2))
+            label_weight += class_count
+
+        # repeat room_idx for num_room_point // num_sample_point times
+        sample_prob = np.array(num_point_all) / float(np.sum(num_point_all))
+        num_iter = int(np.sum(num_point_all) / float(self.num_points))
+        room_idxs = []
+        for idx in range(len(self.data_infos)):
+            room_idxs.extend([idx] * round(sample_prob[idx] * num_iter))
+
+        # calculate label weight, adopted from PointNet++
+        if self.label_weight_func is None:
+            label_weight = np.ones(num_classes, dtype=np.float32)
+        else:
+            label_weight = label_weight[:-1].astype(np.float32)
+            label_weight = label_weight / label_weight.sum()
+            label_weight = self.label_weight_func(label_weight)
+
+        return np.array(room_idxs), label_weight
+
+    def _convert_to_train_label(self, mask):
+        """Convert class_id in loaded segmentation mask to label."""
+        if isinstance(mask, str):
+            if mask.endswith('npy'):
+                mask = np.load(mask)
```

Comment:
Need to careful handle the loading part, and may make it as a loading pipeline to support ceph loading.

Reply:
Add TODO to fix it in the future.